rxandroid/src/main/java/rx/android/lifecycle/LifecycleHelper.java

```diff
@@ -0,0 +1,92 @@
+package rx.android.lifecycle;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import rx.subjects.BehaviorSubject;
+
+/**
+ * This instance lives within the scope of an {@code Activity} in order to forward Lifecycle events
+ * to {@link rx.subjects.BehaviorSubject}
+ *
+ * Thread safe, since it's <i>Thread-confined</i> to the main Thread.
+ *
+ * @author Nabil Hachicha.
+ */
+class LifecycleHelper implements Application.ActivityLifecycleCallbacks {
+    private final Activity activityToMonitor;
+    private final BehaviorSubject<LifecycleEvent> lifecycleSubject;
+    /**
+     * Help track if {@link #onActivityCreated(Activity, Bundle)} was called
+     */
+    private boolean isCreateCalled = false;
+    /**
+     * Help track if {@link #onActivityStarted(Activity)} was called
+     */
+    private boolean isStarted = false;
+
+    public LifecycleHelper(Activity instance, BehaviorSubject<LifecycleEvent> lifecycle) {
```

Comment:
It seems odd to me that you provide the lifecycle object when `LifecycleHelper` is the one that manages it. Why not have `LifecycleHelper` create the `BehaviorSubject`, then you can retrieve it after construction?